packages/dbml-core/src/parse/ANTLR/parsers/mysql/MySqlLexer.g4

```diff
@@ -43,6 +45,19 @@ channels {
     ERRORCHANNEL
 }
 
+// MUST READ: This is used to detect dynamic DELIMITERs, the original grammar does not have this
+@members {
+  this.currentDelimiter = ";";
+
+  this.setDelimiter = function (delimiter) {
+    this.currentDelimiter = delimiter.trim();
+  };
+
+  this.isCurrentDelimiter = function (text) {
+    return text === this.currentDelimiter;
+  };
+}
```

Comment:
This work. Another way is you could define a lexer base class :and move the code here to there. Please check the. For more details, check the PostgresSQL Lexer.g4 and .js for an example